spec/services/framework_nomis_mappings/importer_spec.rb

```diff
@@ -0,0 +1,189 @@
+        created_at: '2013-04-29',
+        expires_at: '2101-07-09',
+        expired: false,
+        active: true,
+        offender_no: 'A9127EK',
+      },
+    ]
+  end
+
+  let(:nomis_reasonable_adjustments) do
+    [
+      {
+        treatment_code: 'DA',
+        comment_text: 'Some comment',
+        start_date: '2014-03-29',
+        end_date: nil,
+        agency_id: 'LGI',
+        treatment_description: 'Some treatment description about DA',
+      },
+      {
+        treatment_code: 'BA',
+        comment_text: 'Some comment',
+        start_date: '2014-03-29',
+        end_date: nil,
+        agency_id: 'AGI',
+        treatment_description: 'Some treatment description about BA',
+      },
+    ]
+  end
+
+  let(:nomis_personal_care_needs) do
+    [
+      {
+        problem_code: 'VI',
+        problem_status: 'ON',
+        problem_description: 'Visually impaired',
+        start_date: '2015-03-29',
+        end_date: '2100-06-08',
+        offender_no: '321',
+      },
+      {
+        problem_code: 'PEEP',
+        problem_status: 'ON',
+        problem_description: 'Special vehicle',
+        start_date: '2015-03-29',
+        end_date: '2100-06-08',
+        offender_no: '321',
+      },
+    ]
+  end
+
+  let(:framework_response1) do
+    alert_code = create(:framework_nomis_code, code: 'VI', code_type: 'alert')
+    question = create(:framework_question, framework_nomis_codes: [alert_code])
+    create(:string_response, framework_question: question)
+  end
+
+  let(:framework_response2) do
+    personal_care_need_code = create(:framework_nomis_code, code: 'VI', code_type: 'personal_care_need')
+    reasonable_adjustment_fallback = create(:framework_nomis_code, code: nil, code_type: 'reasonable_adjustment', fallback: true)
+    personal_care_need_fallback = create(:framework_nomis_code, code: nil, code_type: 'personal_care_need', fallback: true)
+    question = create(:framework_question, framework_nomis_codes: [personal_care_need_code, reasonable_adjustment_fallback, personal_care_need_fallback])
+    create(:string_response, framework_question: question)
+  end
+
+  let(:person) do
+    create(:person, :nomis_synced, latest_nomis_booking_id: 111_111)
+  end
+
+  before do
+    allow(NomisClient::Alerts).to receive(:get).and_return(nomis_alerts)
+    allow(NomisClient::PersonalCareNeeds).to receive(:get).and_return(nomis_personal_care_needs)
+    allow(NomisClient::ReasonableAdjustments).to receive(:get).and_return(nomis_reasonable_adjustments)
+  end
+
+  it 'persists alerts, reasonable adjustments and personal care needs NOMIS mappings' do
+    framework_responses = FrameworkResponse.where(id: [framework_response1.id, framework_response2.id])
+    framework_nomis_codes = framework_responses.flat_map(&:framework_nomis_codes)
+
+    expect { described_class.new(person: person, framework_responses: framework_responses, framework_nomis_codes: framework_nomis_codes).call }.to change(FrameworkNomisMapping, :count).by(6)
```

Comment:
My question before was around why this expects 6 mappings to be created? I'm guessing, perhaps it's because there are two nomis alerts, two reasonable adjustments and two personal care needs? Or maybe something else.. but a little explanation here would be helpful for future readers.

Reply:
Updated spec description makes sense, thanks for addressing 👍